connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -0,0 +1,368 @@
+    }
+
+    /**
+     * Given the topic definition, check whether the topic exists and if not then attempt to create the
+     * missing topic. The brokers will use its default topic options for anything not specified in the supplied definition.
+     * This method returns the topic details for the newly-created or previously-existing topic.
+     * <p>
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topic the specifications of the topic
+     * @return the topic description of the newly-created or existing topic; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public TopicDescription createTopicIfMissing(NewTopic topic) {
+        Map<String, TopicDescription> result = createTopicsIfMissing(topic);
+        return result != null ? result.get(topic.name()) : null;
+    }
+
+    /**
+     * Given the topic definitions, check whether the topics exists and for all those that do not attempt to create the
+     * missing topics. The brokers will use its default topic options for anything not specified in the supplied definitions.
+     * This method returns the topic details for all of the topic included in the supplied descriptions.
+     * <p>
+     * This operation requires the broker to support clients performing several administrative operations. If the broker
+     * does not support this capability, this method returns null.
+     *
+     * @param topics the specifications of the topics
+     * @return the topic description of the newly-created or existing topics, keyed by the topic names; null only
+     * if the broker is too old to perform this operation
+     * @throws ConnectException if an error occurs, the operation takes too long, or the thread is interrupted while
+     *                          attempting to perform this operation
+     */
+    public Map<String, TopicDescription> createTopicsIfMissing(NewTopic... topics) {
+        if (topics == null || topics.length == 0) return Collections.emptyMap();
+
+        Map<String, NewTopic> topicsByName = new HashMap<>();
+        for (NewTopic topic : topics) {
+            if (topic != null) topicsByName.put(topic.name(), topic);
+        }
+        Set<String> allRequestedTopicNames = new HashSet<>(topicsByName.keySet());
+        String bootstrapServers = bootstrapServers();
+
+        try {
+            // Getting topic descriptions for non-existant topics may auto-create them, so first see which of
+            // our topics do already exist
+            Set<String> existingTopicNames = existingTopics(topicsByName.keySet());
+            if (existingTopicNames == null) {
```

Comment:
Yeah, I debated using an exception. 

However, I'm rethinking this logic, and think that I can make it more straightforward and potentially push down some of the topic creation logic into the `AdminClient`, getting rid of the Kafka Connect `TopicAdmin` utility, and further simplifying the Kafka Connect changes. 

The new `TopicAdmin` logic tries to be smart about only creating topics that don't yet exist _and_ to then return a description of all of the topics. I don't think it needs to do all this extra work of listing all of the topics, creating the missing topics, and fetching the descriptions of all the topics. Instead, it should just try to create the topics, handle the `TopicExistsException`, and just return the names of the topics that were created. This is simple enough that we could push this logic down into the `AdminClient` on top of what's already there, and get rid of `TopicAdmin` altogether.

Then `KafkaBasedLog` could instead start up by using the `TopicAdmin` to ensure the topics are created, logging whether the topics were created or already exist. The remainder of `KafkaBasedLog` can be what it used to be, including setting up the consumers as before, since the topics should already be created at that point.

I'll put together a commit that makes these changes, and can work with @cmccabe to see if the changes to `AdminClient` would be acceptable.